lib/intercom_stats/intercom/worker.ex

```diff
@@ -175,25 +168,47 @@ defmodule IntercomStats.Intercom.Worker do
     end
   end
 
-  defp calculate_closing_time(%{"created_at" => created}, close_timestamp, snooze_time) do
-    close_timestamp - created - snooze_time
-  end
-
-  defp determine_snooze_time(%{"conversation_parts" => %{"conversation_parts" => parts}}) do
+  defp determine_open_time(%{
+         "conversation_parts" => %{"conversation_parts" => parts},
+         "created_at" => created_at
+       }) do
     {result, _} =
       Enum.flat_map_reduce(parts, %{}, fn i, acc ->
-        with "snoozed" <- acc["part_type"] do
-          {[calculate_snooze_time(i, acc)], i}
-        else
-          _ -> {[], i}
+        cond do
+          is_not_open(i) and acc == %{} ->
+            {[i["created_at"] - created_at], i}
+
+          is_not_open(i) and is_open(acc) ->
+            {[calculate_open_time(i, acc)], i}
+
+          is_open(i) and is_not_open(acc) ->
+            {[], i}
+
+          true ->
+            {[], acc}
         end
       end)
 
     Enum.sum(result)
   end
 
-  defp calculate_snooze_time(part, snoozed_part) do
-    part["created_at"] - snoozed_part["created_at"]
+  defp is_open(part) do
```

Comment:
not necessary, but a nice convention in elixir is to use function names with `?` to denote that the function returns a boolean. In this case, `open?(...)`